runtime/runtime/src/actions.rs

```diff
@@ -445,7 +446,28 @@ pub(crate) fn action_delete_account(
     result: &mut ActionResult,
     account_id: &AccountId,
     delete_account: &DeleteAccountAction,
+    current_protocol_version: ProtocolVersion,
 ) -> Result<(), StorageError> {
+    if current_protocol_version
+        >= STABLE_PROTOCOL_FEATURES_TO_VERSION_MAPPING[&ProtocolFeature::DeleteActionRestriction]
+    {
+        let account = account.as_ref().unwrap();
+        let mut account_storage_usage = account.storage_usage;
+        let contract_code = get_code(state_update, account_id, Some(account.code_hash))?;
+        if let Some(code) = contract_code {
+            // account storage usage should be larger than code size
+            let code_len = code.code.len() as u64;
+            debug_assert!(account_storage_usage > code_len);
+            account_storage_usage -= code_len;
```

Comment:
Considering previous issue with state size due to delete key action, we should use `saturated_sub` to be safe